### DISCUSSION TITLE
Web UI

### DISCUSSION THREAD

**Discussion by drewp** [pyinfra-dev/pyinfra#1062] [Ideas]
Here's a mockup of what I'm thinking of writing:
![image](https://github.com/pyinfra-dev/pyinfra/assets/104003/70ce6509-c28b-425e-97d7-c2315bcb4664)

This is designed for my own use case of 10 hosts; 1600 lines of py config in 20 files; frequent enabling/disabling of hosts and files.

Suggestions and clarifications welcome.

**Comment by goetzk** [comment]
Hi @drewp , did you end up developing this?

**Comment by drewp** [comment]
It's still on my list, so to speak :)
Are you interested in designing, building, or using it?

**Reply by goetzk** [reply]
I'm still considering pyinfra and made some notes last week about what I'd want in a web UI - if you had written one that would have been interesting to look at :)
If I go with pyinfra and a web ui exists, I'll be interested in using (and likely debugging) it.

**Reply by Firesphere** [reply]
Not sure if I can/am able to contribute to building/designing/architecting (lack of time), but definitely use!

A good start would probably be, instead of all the tickboxes etc., to have a process of something like this:
-> Dropdown/multiselect "Select server(s)"
-> Single-select/Radio "Select task" (main/code-only/etc.)
-> Run

Output, a list, just waht PyInfra returns already. Maybe some clarifications, or possibly "time it took"?
